src/applications/maliput_to_dot.cc

```diff
@@ -0,0 +1,123 @@
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+/// @file maliput_to_dot.cc
+///
+/// Builds a dragway, multilane or malidrive road geometry
+/// and creates a routing::graph::Graph from it to then serialize to a file a DOT representation.
+///
+/// @note
+/// 1. It allows to create an OBJ file from different road geometry implementations.
+///     The `maliput_backend` flag will determine the backend to be used.
+///    - "dragway": The following flags are supported to use in order to create dragway road geometry:
+///       -num_lanes, -length, -lane_width, -shoulder_width, -maximum_height.
+///    - "multilane": yaml file path must be provided:
+///       -yaml_file.
+///    - "malidrive": xodr file path must be provided and the tolerance is optional:
+///         -xodr_file_path -linear_tolerance.
+/// 2. The applications possesses flags to modify the DOT file builder:
+///      -dot_dir_path, -dot_file_name
+/// 3. The level of the logger could be setted by: -log_level.
+
+#include <fstream>
+#include <limits>
+#include <string>
+
+#include <gflags/gflags.h>
+#include <maliput/common/filesystem.h>
+#include <maliput/common/logger.h>
+#include <maliput/common/maliput_abort.h>
+#include <maliput/routing/graph/graph.h>
+#include <maliput/utility/generate_dot.h>
+
+#include "integration/tools.h"
+#include "maliput_gflags.h"
+
+COMMON_PROPERTIES_FLAGS();
+MULTILANE_PROPERTIES_FLAGS();
+DRAGWAY_PROPERTIES_FLAGS();
+MALIDRIVE_PROPERTIES_FLAGS();
+MALIPUT_OSM_PROPERTIES_FLAGS();
+MALIPUT_APPLICATION_DEFINE_LOG_LEVEL_FLAG();
+
+DEFINE_string(maliput_backend, "dragway", "Whether to use <dragway>, <multilane> or <malidrive>. Default is dragway.");
+
+// Gflags for output files.
+DEFINE_string(dot_dir_path, ".", "Directory to contain DOT file.");
+DEFINE_string(dot_file_name, "maliput_graph.dot", "Name of the maliput::routing::graph::Graph DOT file.");
+
+namespace maliput {
+namespace integration {
+namespace {
+
+// Generates an OBJ file from a YAML file path or from
```

Comment:
nit: Remove the occurrences of "obj" in the file

Reply:
[ac6cf48](https://github.com/maliput/maliput_integration/pull/139/commits/ac6cf48faf346a4b77839be669d8e6fba0175211)